### ISSUE TITLE
A quick question about obtaining specific element array

### ISSUE THREAD

**Issue by sungho91** [mfem/mfem#3651]
Hi all, 

I want to refine mesh locally at an element attribute. And, I found "GeneralRefinement" which can locally refine chosen elements. So, I'd like to get an element array corresponding to element attributes.

Do we have a function that can handle this?

Best

Sungho Lee

**Comment by v-dobrev** [comment]
You can do this is two easy steps: (1) construct an `Array<int>` that defines the mapping element -> attribute (use `Mesh::GetAttribute` to fill-in the array); (2) from this array you can then construct the transpose mapping, attribute -> elements, as a `Table` using this function https://github.com/mfem/mfem/blob/5de386b084b4d5bc10a284664bf11b18f0afd81e/general/table.hpp#L210-L211

To access the list of all elements with a given attribute `attr`, you can use the following methods with `attr` as the parameter `i`: https://github.com/mfem/mfem/blob/5de386b084b4d5bc10a284664bf11b18f0afd81e/general/table.hpp#L105-L111

**Comment by sungho91** [comment]
Hi @v-dobrev,

Thank you for your comment and it will be helpful. I'm gonna look into it. 
By the way, I found my own way to solve the problem. I looped through all the elements and checked the attribute of each element. Then, I saved a vector corresponding to the element attributes. The following lines show what I did:

// Local refinement
Array<int> refs;
for (int i = 0; i < mesh->GetNE(); i++)
{
if(mesh->GetAttribute(i) == 2)
{
refs.Append(i);
}
}

mesh->GeneralRefinement(refs, 1).

Many thanks,
Sungho
